conjure_python_client/_http/requests_client.py

```diff
@@ -196,18 +201,43 @@ def create(
 class TransportAdapter(HTTPAdapter):
     """Transport adapter that allows customising ssl things"""
 
+    def __init__(self, *args, enable_keep_alive: bool = False, **kwargs):
+        self._enable_keep_alive = enable_keep_alive
+        super().__init__(*args, **kwargs)
+
     def init_poolmanager(
         self, connections, maxsize, block=False, **pool_kwargs
     ):
         self._pool_connections = connections
         self._pool_maxsize = maxsize
         self._pool_block = block
         ssl_context = create_urllib3_context(ciphers=CIPHERS)
+        keep_alive_pool_kwargs = {
+            "socket_options": HTTPConnection.default_socket_options
+            + [
+                (
+                    socket.SOL_SOCKET,
+                    socket.SO_KEEPALIVE,
+                    1,
+                ),  # Enable keep alive.
+                (
+                    socket.SOL_TCP,
+                    socket.TCP_KEEPIDLE,
+                    120,
+                ),  # After 120s of idle connection, start keepalive probes.
+                (
+                    socket.SOL_TCP,
+                    socket.TCP_KEEPINTVL,
+                    120,
+                ),  # Interval of 120s between individual keepalive probes.
+            ]
+        }
+        if self._enable_keep_alive:
+            pool_kwargs = {**pool_kwargs, **keep_alive_pool_kwargs}
         self.poolmanager = PoolManager(
             num_pools=connections,
             maxsize=maxsize,
             block=block,
-            strict=True,
```

Comment:
why?

Reply:
This option is no longer necessary, warning thrown:
`  /home/circleci/project/.tox/py3/lib/python3.8/site-packages/urllib3/poolmanager.py:315: DeprecationWarning: The 'strict' parameter is no longer needed on Python 3+. This will raise an error in urllib3 v2.1.0.`